### PR TITLE
Proof of Concept

### PR DESCRIPTION
**Pull request by rissson**

### CODE REVIEW

**Comment by rissson** [comment]
I don't know why github wants me to rebase this.

**Comment by rissson** [comment]
Also, I'd like #6 to be merged first so I can rebase this on top of it and this work can also be licensed under my name.

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+      pkgImport = pkgs: system:
+        import pkgs {
+          inherit system;
+        };
+
+      multiSystemOutputs = eachDefaultSystem (system:
+        let
+          pkgs = pkgImport nixpkgs system;
+        in
```
Comment (kalbasit) [inline flake.nix:27]:
```suggestion
      multiSystemOutputs = eachDefaultSystem (system:
        let
          pkgs = import nixpkgs { inherit system; };
        in
```

File: lib/nixos-system.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    self.nixosModules.soxin
+
+    home-manager.nixosModules.home-manager
+    {
+      # This is required when using flakes.
+      home-manager.useGlobalPkgs = true;
+      home-manager.useUserPackages = true;
+    }
+
+    ({ config, ... }: {
```
Comment (kalbasit) [inline lib/nixos-system.nix:21]: Add a comment as to why this is necessary.

File: modules/default.nix [hunk 3]
```diff
@@ -0,0 +1 @@
+builtins.mapAttrs (_: moduleFile: import moduleFile) (import ./list.nix)
```
Comment (kalbasit) [inline modules/default.nix:1]: Although this is fine, we should adopt a conditional import in the future so we can import or not import files based on some condition (for instance no xorg on Darwin.)

**Review by kalbasit — APPROVED** [review]

File: modules/default.nix (hunk 3 above)
Comment (rissson) [reply modules/default.nix:1]: That would conflict with the way flakes work. If a user only wants to use some modules, they can choose to do so. Besides, we can always choose what we activate with the `mode` argument.

File: flake.nix (hunk 1 above)
Comment (rissson) [reply flake.nix:27]: I think that since `nixpkgs` is only used here we can do something like `nixpkgs.${system}.legacyPackages`

**Pull request merged**